FABulous.py

```diff
@@ -0,0 +1,743 @@
+# Copyright 2021 University of Manchester
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from contextlib import redirect_stdout
+from fabric_generator.utilities import genFabricObject, GetFabric
+import fabric_generator.model_generation_vpr as model_gen_vpr
+import fabric_generator.model_generation_npnr as model_gen_npnr
+from fabric_generator.code_generation_VHDL import VHDLWriter
+from fabric_generator.code_generation_Verilog import VerilogWriter
+import fabric_generator.code_generator as codeGen
+import fabric_generator.file_parser as fileParser
+from fabric_generator.fabric import Fabric, Tile
+from fabric_generator.fabric_gen import FabricGenerator
+import csv
+from glob import glob
+import os
+import argparse
+import pickle
+import re
+import sys
+import subprocess as sp
+import shutil
+from typing import List, Literal
+import docker
+import cmd
+import readline
+import logging
+readline.set_completer_delims(' \t\n')
+
+fabulousRoot = os.getenv('FAB_ROOT')
+if fabulousRoot is None:
+    print('FAB_ROOT environment variable not set!')
+    print("Use 'export FAB_ROOT=<path to FABulous root>'")
+    sys.exit(-1)
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(
+    format="[%(levelname)s]-%(asctime)s - %(message)s", level=logging.INFO)
+
+
+# Create a FABulous Verilog project that contains all the required files
+def create_project(project_dir, type: Literal["verilog", "vhdl"] = "verilog"):
+    if os.path.exists(project_dir):
+        print('Project directory already exists!')
+        sys.exit()
+    else:
+        os.mkdir(f"./{project_dir}")
```

Comment:
appending `./` to the `project_dir` is definitely not right if an absolute project dir is specified - I don't think it's needed at all